source/mqtt/Mqtt5ClientCore.cpp

```diff
@@ -642,6 +609,43 @@ namespace Aws
                 }
             }
 
+            Mqtt5to3AdapterOptions::Mqtt5to3AdapterOptions() {}
+
+            ScopedResource<Mqtt5to3AdapterOptions> Mqtt5to3AdapterOptions::NewMqtt5to3AdapterOptions(
+                const Mqtt5ClientOptions &options) noexcept
+            {
+                Allocator *allocator = options.m_allocator;
+                ScopedResource<Mqtt5to3AdapterOptions> adapterOptions = ScopedResource<Mqtt5to3AdapterOptions>(
+                    Crt::New<Mqtt5to3AdapterOptions>(allocator),
+                    [allocator](Mqtt5to3AdapterOptions *options) { Crt::Delete(options, allocator); });
+                adapterOptions->m_mqtt3options.allocator = options.m_allocator;
+                adapterOptions->m_mqtt3options.hostName = options.m_hostName.c_str();
```

Comment:
Storing a pointer to the underlying data of `m_hostName` can cause a use-after-free issue. This code is moved from `Mqtt5ClientCore::NewConnection` as is, but I think it should be addressed in this PR.
Looking at the other members, it seems the issue is with `m_hostName` only, but it's better to check them as well.